mcp/gcp_ae.tf

```diff
@@ -211,14 +179,12 @@ resource "google_app_engine_flexible_app_version" "self" {
       min_idle_instances = lookup(automatic_scaling.value, "min_idle_instances", null)
       min_pending_latency = lookup(automatic_scaling.value, "min_pending_latency", null)
       min_total_instances = lookup(automatic_scaling.value, "min_total_instances", null)
-
-      dynamic "cpu_utilization" {
-        # cpu_utilization is a required block so doesn't need a condition like other dynamic blocks
-        for_each = lookup(automatic_scaling.value, "cpu_utilization", {cpu_utilization: {target_utilization: local.default.automatic_scaling.target_utilization}})  # required default. Also see attribute below
-
+      dynamic "cpu_utilization"{
+        for_each = lookup(automatic_scaling.value, "cpu_utilization", null) == null ? {cpu_utilization: {target_utilization: local.default.automatic_scaling.target_utilization}} : {cpu_utilization: automatic_scaling.value.cpu_utilization}
```

Comment:
This would originally lead to errors if user specified `cpu_utlization` values. as it would be a map of:
```
{
  target_utilization = 0.5
  aggregation_window_length = 10s
}
```
instead of:
```
{
  cpu_utilization : {
      target_utilization = 0.5
      aggregation_window_length = 10s
  }
}
```
So changed to make it the latter

Reply:
If it's null, that works as well? I thought I remembered `for_each` not liking null

Reply:
If it is null it will use the first option of `{cpu_utilization: {target_utilization: local.default.automatic_scaling.target_utilization}}` to return a map with the default required values.  
The error was only for when the user specified `cpu_utilization`. The map format was wrong and would mean the `for_each` would be taking `target_utilization` and `aggregation_window_length` as the keys to iterate through, rather than the object containing them as values